doonut-core/src/main/java/com/doonutmate/doonut/member/model/OauthTypeStrategy.java

```diff
@@ -0,0 +1,12 @@
+package com.doonutmate.doonut.member.model;
+
+public enum OauthTypeStrategy {
+    KAKAO("kakaoMemberDelete"),
+    APPLE("appleMemberDelete");
```

Comment:
`KakaoMemberDelete.kt`, `AppleMemberDelete.kt` 클래스명이 바뀐다면 여기도 변경해줘야하는데, 이건 런타임 에러 때 발견가능해서 언제든 터질지 모르는 시한폭탄인 것 같아요.  

인터페이스에 추상 프로퍼티를 만들어서 개선해볼까요?  

```suggestion
public enum OauthTypeStrategy {
    val oauthType: OauthType
```

Reply:
OauthType하고 OauthTypeStrategy을 하나로 합칠 수 있지 않을까요?

Reply:
제가 생각했을때 oauthType이 현재 시스템에서 전반적으로 사용되는데, 만약에 지금처럼 oauthType에 strategyName을 추가를 해버리면 기존 로직에서 필요없는 strategyName을 가지게 되는 문제가있고 추후에 기존 코드가 바뀔때 문제가 될수도있어 책임을 분리하고자 했습니다!

Reply:
하나로 합친다는 의미를 저는 두개의 enum이 사실상 중복이여서 하나로 사용하는 방향이 어떤지로 이해를 하였는데 맞을까요??
바로 위에서 설명한 이유때문에 enum을 분리하였는데 혹시 하나로 합치는게 더 좋은 방향일까요??
일단은 enum명에 strategy를 제외하는 것에는 공감하여 OauthTypeStrategy을 MemberDeleteOauthType로 이름으로 변경하였습니다.

Reply:
이제 OauthTypeStrategy에 strategyName 필드 없애도 되지 않나용?

Reply:
> 하나로 합친다는 의미를 저는 두개의 enum이 사실상 중복이여서 하나로 사용하는 방향이 어떤지로 이해를 하였는데 맞을까요??

맞습니다!

Reply:
이해했습니다. 애초에 읽을때 값을 반환하게 해뒀는데 state를 enum에 저장하고 있는게 이상했네요... 코드를 잘못 이해하고 사용한거같네요 🥲

enum이 값을 안가지게 되어서 기존 새로 생성한 enum도 삭제하고 기존 oauthType으로 사용하는 것으로 하겠습니다!